src/parameterizations/vertical/MOM_opacity.F90

```diff
@@ -1111,11 +1152,43 @@ subroutine opacity_init(Time, G, GV, US, param_file, diag, CS, optics)
       optics%max_wavelength_band(2)=700
     endif
     if (optics%nbands > 2) then
+      I_NIR_bands = 1.0 / real(optics%nbands - 2)
       do n=3,optics%nbands
-        optics%min_wavelength_band(n) =700
-        optics%max_wavelength_band(n) =2800
+        optics%min_wavelength_band(n) = 700. + (n-3)*2100.0*I_NIR_bands
```

Comment:
should hardcoded values also be parameters?

Reply:
Yes, they should, and the PR has now been modified so that they are runtime parameters.